Adds opengraph partial to slim header

matt-bernhardt opened the pull request:
## Developer

### Why are these changes being introduced:

There are two current problems with our use of OpenGraph headers on the WordPress network:
1. The themes don't currently always include the headers, because one template doesn't call for that partial.
2. The `og_description` tag uses the default behavior by WordPress, which is to truncate the description at 55 words.

As a result of these changes, our attempts to harvest site content in TIMDEX are hamstrung in some cases (content that uses that header, and content that is very long).

### Relevant ticket(s):

https://mitlibraries.atlassian.net/browse/pw-60

### How does this address that need:

1. The slim header which previously did not include the opengraph partial now does.
2. We define a custom length for the description of 100 words - (instead of the initially-proposed 9,999 words).

### Document any side effects to this change:

* These changes make our pages heavier in bandwidth terms, because we are including two copies of the body in every HTML response.
* There may yet be other needed changes to our OpenGraph implementation, depending on the evolving needs of the TIMDEX harvester - but that is more accurately described as "future work to do" or "problems not yet solved" rather than "side effects caused by this change".

### Stylesheets

- [x] Any theme or plugin whose stylesheets have changed has had its version
      string incremented.

### Secrets

- [ ] All new secrets have been added to Pantheon tiers
- [ ] Relevant secrets have been updated in Github Actions
- [ ] All new secrets documented in README
- [x] No secrets are implicated by this change

### Documentation

- [ ] Project documentation has been updated
- [x] No documentation changes are needed

### Accessibility

- [ ] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)

### Stakeholder approval

- [x] Stakeholder approval has been confirmed
- [ ] Stakeholder approval is not needed

### Dependencies

NO dependencies are updated


## Code Reviewer

- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [ ] The changes have been verified
- [ ] The documentation has been updated or is unnecessary
- [ ] New dependencies are appropriate or there were no changes

matt-bernhardt committed:
Adds opengraph partial to slim header

** Why are these changes being introduced:

* We have realized that sites which elect to use the slim header
  end up not having opengraph tags in their page head, because the
  slim header partial does not include a call to the opengraph partial.

** Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/pw-60

** How does this address that need:

* This adds a call to the opengraph partial to the slim header template,
  which should result in all WordPress pages now featuring OpenGraph
  tags.

** Document any side effects to this change:

* None that I can detect.

matt-bernhardt committed:
Define custom excerpt length

** Why are these changes being introduced:

* WordPress has a default limit for page excerpts of 55 words, broken on
  a word boundary. This is frequently too short of a snippet to give a
  good indication of page content.

** Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/pw-60

** How does this address that need:

* This defines a custom function to extend the length of the excerpt to
  9,999 words - which is longer than any page on our site should be.
  With this change, the opengraph tag `og:description` in our site
  headers should be a copy of all text in the page body.

** Document any side effects to this change:

* This makes the notion of an "excerpt" kind of meaningless, as it just
  repeats the full body copy.

JPrevost commented on web/app/themes/mitlib-parent/functions.php:
```diff
@@ -494,6 +494,17 @@ function customize_body_class( $classes ) {
+ * @return int The maximum number of words to include from the excerpt.
+ */
+function custom_excerpt_length( $length ) {
+	return 9999;
```
Hmmm. Was this high value determined for TIMDEX purposes or did it already exist general purposes? It feels questionable at glance to just dump _all_ of the content of a page into a summary-like field and call it done.

matt-bernhardt replied:
This size was chosen as a value that should be big enough for no pages to be truncated. Graham and I have looked at the result of a crawl with this in place (in TIMX-253 ), and found that the longest page we have is currently 9,066 words (second longest is ~5,600 words).

Without this code change, the description field would be truncated at 55 words - so there are two decisions being made here:
1. We should change the description length away from its present value of 55 words, and
2. The description length should be large enough that no page is truncated.

I definitely think that we should not be using 55 words for this value. In terms of what the proper value is, I'm happy to discuss whether we should be imposing some sort of length limit. However, I think the preferrable setup at that point would be to actually provide a summary in the description field, rather than a truncated copy of the content.

JPrevost replied:
Gotcha. I have to sit with this a bit more. I definitely agree 55 is not the correct number so "bigger than 55" I'm on board with. I'll likely want to chat synchronously on this next week to better understand our intent and options. Thanks for the extra details!

JPrevost replied:
@matt-bernhardt which specific opengraph tag are we populating that is affected by this change?

matt-bernhardt replied:
The affected tag is `og:description` - there's a link in https://mitlibraries.atlassian.net/browse/TIMX-253 to the most-affected page ( "Signers of the 1861 act of association" on the MIT History site)

JPrevost reviewed (changes requested):
@matt-bernhardt can you revert the change to extend the summary to effectively be duplicative of the full content body and just make this PR to add the OG tags to the slim header pages? I'm still open to discuss this, but based on what that tag was intended to mean our change here is not appropriate is my interpretation.

matt-bernhardt committed:
Set description length limit to 100 words

** Why are these changes being introduced:

* During code review we discussed whether 9,999 words (effectively "no
  limit") was the right size for the og:description field, and decided
  that it was not.

** Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/pw-60

** How does this address that need:

* This sets the maximum length of the description to be 100 words. This
  limit was chosen arbitrarily to be a round number that is larger than
  55 (the default value), while still being overwhelmingtly long.

** Document any side effects to this change:

* None - although as noted above the 100-word length limit was chosen
  somewhat at random.

matt-bernhardt commented:
Okay @JPrevost and @ghukill - I've pushed an update to this work that sets the maximum length of `og:description` to be 100 words. I think this is a good choice for the moment - if we get to the point of writing these descriptions ourselves (rather than just auto-generating by truncating the page content), then we'll be in a better position to pick a meaningful limit.
